### DISCUSSION TITLE
Allow automerge to apply to existing Renovate MRs

### DISCUSSION THREAD

**Discussion by TimKnight-DWP** [renovatebot/renovate#19331] [Request Help] [answered]
### What would you like Renovate to be able to do?

If we have open Renovate MRs/branches, especially if doing testing on regexManagers and packageRules, it would be ideal if when we are happy with them and enable automerge, then there is an additional option available to request renovate to automerge existing MRs on those branches.

Currently when re-renovating a branch it detects no changes required, it realises it wants to make an automerge MR, but when it pushes that to GitLab it gets a 4xx error because an MR already exists for that branch

### If you have any ideas on how this should be implemented, please tell us here.

If automergeOnOldMRs (or whatever this new option might be) option enabled, Renovate will get existing MRs on a branch and apply automerge labels/tags etc and call attemptAutomergePr

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
I don't understand everything you're saying here, so please clarify my questions below:

> If we have open Renovate MRs/branches, especially if doing testing on regexManagers and packageRules, it would be ideal if when we are happy with them and enable automerge, then there is an additional option available to request renovate to automerge existing MRs on those branches.

Here you are asking for functionality such as a checkbox to tick in the MR body which would mean "please automerge this MR even though it does not have automerge enabled in Renovate config"?

> Currently when re-renovating a branch 

What do you mean by "re-renovating"?

> it detects no changes required, it realises it wants to make an automerge MR,

Realizes it how? Have you changed your Renovate config to enable automerge?

> but when it pushes that to GitLab it gets a 4xx error because an MR already exists for that branch

This sounds like another problem, such as perhaps you have an incorrectly configured username so Renovate does not "find" the existing MR.

> If automergeOnOldMRs (or whatever this new option might be) option enabled, Renovate will get existing MRs on a branch and apply automerge labels/tags etc and call attemptAutomergePr

Today if you change config and enable automerge after creation, I would have expected Renovate to already try (itself)

**Comment by TimKnight-DWP** [comment]
Sorry poor communication on my part.

Scenario was:

We added a new configuration to our packageRules, to group some specific file changes together on a specific branch (due to branch name configuration internal to DWP).

We run renovate across all repos every night, so had MRs generated for this new packageRule in the morning.

Upon verifying the MRs looked good we enabled automerge: true, platformAutomerge: true and ignoreGitlabApprovals: true.

I ran the updated configuration against a repo with one of the existing MRs from the previous night, and could see in the logs that no additonal updates were required on the branch (because no new versions had been published), and that renovate tried to create a new MR in Gitlab with the automerge rules enabled. It failed to create that new MR because there already existed an MR on the branch, and it did not try to automerge the existing MR despite that feature now being enabled

**Comment by viceice** [comment, marked as answer]
you've probably a username mismatch, so renovate don't find its own existing PR. did the PR title change when automerge is enabled?

**Reply by TimKnight-DWP** [reply]
Aaaaaah, when we testing I'd have had to use a PAT of mine but on it's main schedule run renovate uses it's own GAT.

I'll double check to see if it applied the new automerge on overnight runs, thank you

**Reply by TimKnight-DWP** [reply]
Perfect I can see the overweekend runs with the GAT applied automerge and attempted 👍

**Comment by rarkins** [comment]
Converted to a discussion as it sounds like you have a config problem

**Comment by TimKnight-DWP** [comment]
Understand how the config works now thank you @viceice 

Feel free to close
